src/hooks/bookmarks/use-is-bookmarked.ts

```diff
@@ -0,0 +1,42 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query'
+import { Tutorial } from 'lib/learn-client/types'
+import { useAllBookmarks } from './use-all-bookmarks'
+
+interface UseIsBookmarkedOptions {
+	tutorialId: Tutorial['id']
+}
+
+interface UseIsBookmarkedResult extends Omit<UseQueryResult, 'data'> {
+	isBookmarked: null | boolean
+}
+
+/**
+ * Handles checking if there is a bookmark for the given `tutorialId`.
+ */
+const useIsBookmarked = ({
+	tutorialId,
+}: UseIsBookmarkedOptions): UseIsBookmarkedResult => {
+	// Load up all bookmarks
+	const { bookmarks, isFetched: bookmarksAreFetched } = useAllBookmarks()
```

Comment:
I don't think the `select` option is what we want, because it'd cause the `find` to be invoked every time the hook renders. I wrapped the `find` in a `useQuery` so that wouldn't happen, and the value would be cached and accessible anywhere.

But I'll go back to the query priming and see if I can get that working as we want it to.

Reply:
I don't expect that the usage of `find` here will have any meaningful performance impact in practice. If we were noticing performance issues, the selector can also be optimized to ensure it only runs when the underlying data changes, per the linked post!

I can see some benefit to the unique queries per bookmark though, considering we'll need to invalidate them on mutations. 💭